### PR TITLE
Don't link Python extensions to libpython

### PR DESCRIPTION
**Pull request by rshin**
It shouldn't be necessary, and might make the extension modules stop working when the Python binary is updated.

### CODE REVIEW

File: src/common/cmake/Common.cmake [hunk 1]
```diff
@@ -43,46 +43,21 @@ message(STATUS "PYTHON_LIBRARY_NAME: " ${PYTHON_LIBRARY_NAME})
...
 # work.
-if(PYTHON_LIBRARIES AND PYTHON_INCLUDE_DIRS)
+if(PYTHON_INCLUDE_DIRS)
   message(STATUS "The custom approach for finding Python succeeded.")
   SET(PYTHONLIBS_FOUND TRUE)
 else()
   message(WARNING "The custom approach for finding Python failed. Defaulting to find_package.")
   find_package(PythonInterp REQUIRED)
-  find_package(PythonLibs ${PYTHON_VERSION_STRING} EXACT REQUIRED)
+  find_package(PythonLibs ${PYTHON_VERSION_STRING} EXACT)
```
Comment (robertnishihara) [inline src/common/cmake/Common.cmake]: do we still need to find the PythonLibs here?

Comment (rshin) [inline src/common/cmake/Common.cmake]: Yes, I believe it's needed to set PYTHON_INCLUDE_DIRS.

**Review by pcmoritz — APPROVED** [review]
Looks great! I really like this improvement. Also tested it on system Python 3 with ubuntu.

**Comment by robertnishihara** [comment]
Nice job! It's great to get rid of a bunch of this code. These lines in the CMakelists for finding the "right" Python libraries went through many painful iterations.

**Pull request merged**